datafusion/expr/src/type_coercion/functions.rs

```diff
@@ -177,35 +177,12 @@ pub fn can_coerce_from(type_into: &DataType, type_from: &DataType) -> bool {
         Timestamp(TimeUnit::Nanosecond, None) => {
             matches!(type_from, Null | Timestamp(_, None))
         }
-        Date32 => {
-            matches!(type_from, Null | Timestamp(_, None))
-        }
         Utf8 | LargeUtf8 => true,
         Null => can_cast_types(type_from, type_into),
         _ => false,
     }
 }
 
-/// Returns a common coerced datatype between 2 given datatypes
-///
-/// See the module level documentation for more detail on coercion.
-pub fn get_common_coerced_type(
```

Comment:
I think this method can be of some interest, if the user wants to check the common coerced type between 2 datatypes?

Reply:
It may be worth waiting for a specific use before leaving it in 🤔

Reply:
> I think this method can be of some interest, if the user wants to check the common coerced type between 2 datatypes?

the common coerced type will be decided by `input exprs/input types` and `op`, if the `op` is different and the same `input types/input exprs` may get the different coerced type.

Reply:
Let's merge this PR then -- we can always recover `get_common_coerced_type` from this PR or git history if we need it.